modules/holodeckb2b-core/src/main/java/org/holodeckb2b/ebms3/headervalidators/PullRequest.java

```diff
@@ -49,7 +49,7 @@ public static InvalidHeader validate(final IPullRequest pullReqInfo) {
         errDetails.append(MessageUnit.validate(pullReqInfo));
 
         // Check that a RefToMessageId is included
```

Comment:
Actually the problem is in this comment which is incorrect :-(  It should have read _"Check that **no** RefToMessageId is included"_
Therefore the original if statement was correct.